### PR TITLE
Test session error handler with raising fake error

### PR DESCRIPTION
**Pull request by desistefanova**
This PR is only for code review. Probably will not be merged.
Depends on the fix in #588

### CODE REVIEW

File: src/sync_session.cpp [hunk 1]
```diff
@@ -125,4 +128,18 @@ RLM_API uint64_t realm_dart_sync_session_register_connection_state_change_callba
 
 } // anonymous namespace
 
+RLM_API void realm_syncsession_report_error_for_testing(realm_sync_session_t* session, uint32_t category, int errorCode, bool isFatal) noexcept
```
Comment (blagoev) [inline src/sync_session.cpp:131]: we prefix our native methods with `realm_dart_`

Comment (blagoev) [inline src/sync_session.cpp:131]:
```suggestion
RLM_API void realm_dart_syncsession_report_error_for_testing(realm_sync_session_t* session, uint32_t category, int errorCode, bool isFatal) noexcept
```

File: lib/src/session.dart [hunk 2]
```diff
@@ -71,6 +71,10 @@ class Session {
     final controller = SessionProgressNotificationsController(this, direction, mode);
     return controller.createStream();
   }
+
+  void raiseSessionError(SyncErrorCategory category, int errorCode, bool isFatal) {
```
Comment (blagoev) [inline lib/src/session.dart:75]: this should be an internal method.

**Review by blagoev — CHANGES_REQUESTED** [review]

**Comment by desistefanova** [comment]
I'm closing this PR, because it is already merged into [Sync config error handler #577](https://github.com/realm/realm-dart/pull/577)

File: src/sync_session.cpp (hunk 1 above)
Comment (desistefanova) [reply src/sync_session.cpp:131]: Done in https://github.com/realm/realm-dart/pull/577

File: lib/src/session.dart (hunk 2 above)
Comment (desistefanova) [reply lib/src/session.dart:75]: Done in https://github.com/realm/realm-dart/pull/577

**Pull request closed without merging**